patch to opendkim.c:5023:12

```
opendkim.c:5023:12: warning: comparison of array 'dfc->mctx_domain' equal to a null pointer is always false [-Wtautological-pointer-compare]
                    dfc->mctx_domain == NULL)
                    ~~~~~^~~~~~~~~~~    ~~~~
```

I think you mean this instead:

    -                   dfc->mctx_domain == NULL)
    +                   strcmp( dfc->mctx_domain, "&" ) == 0)

--- a/opendkim/opendkim.c
+++ b/opendkim/opendkim.c
@@ -4977,7 +4977,7 @@ dkimf_add_signrequest(struct msgctx *dfc, DKIMF_DB keytable, char *keyname,
 		}
 
 		if (domain[0] == '%' && domain[1] == '\0' &&
-		    dfc->mctx_domain == NULL)
+		    dfc->mctx_domain[0] == '\0')
 		{
 			if (dolog)
 			{

